builtin/providers/azure/utils.go

```diff
@@ -0,0 +1,37 @@
+package azure
+
+import (
+	"math/rand"
+	"strings"
+)
+
+// reverseDNSName simply reverses the provided DNS name.
+func reverseDNSName(dnsName string) string {
+	bits := strings.Split(dnsName, ".")
+	// reverse the bits:
+	for i, j := 0, len(bits)-1; i < j; i, j = i+1, j-1 {
+		bits[i], bits[j] = bits[j], bits[i]
+	}
+	return strings.Join(bits, ".")
+}
+
+// getRandomStringLabel returns a random string of the given length.
+func getRandomStringLabel(n int) string {
+	var chars = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+	buf := make([]rune, n)
+	for i := 0; i < n; i++ {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return string(buf)
+}
```

Comment:
Yeah; I thought this was a better approach for clearly identifying the resources internally; but considering that literally all of the resources on Azure are uniquely-named [within their resource type], and that (from what I can tell) the way terraform works; different resource types can't have ID clashes happen, I decided a couple of days ago that @svanharmelen approach of just using the names for the ID's is more sane from all points of view; so fixed.